Sources/MapboxMaps/Location/Pucks/LocationPuckManager.swift

```diff
@@ -49,19 +56,24 @@ internal class LocationPuckManager: LocationConsumer {
     /// The current puck type
     internal private(set) var puckType: PuckType
 
+    /// The type of value that should be passed for bearing
+    internal var bearingType: BearingType
+
     internal init(locationSupportableMapView: LocationSupportableMapView,
                   style: LocationStyleDelegate?,
-                  puckType: PuckType) {
+                  puckType: PuckType,
+                  bearingType: BearingType) {
         puckStyle = .precise
         self.puckType = puckType
+        self.bearingType = bearingType
         self.locationSupportableMapView = locationSupportableMapView
         self.style = style
     }
 
     /// LocationConsumer protocol method that will handle location updates
     internal func locationUpdate(newLocation: Location) {
         if let puck = self.puck {
-            puck.updateLocation(location: newLocation)
+            puck.updateLocation(location: newLocation, bearingType: bearingType)
```

Comment:
feels like overkill that you're passing in the `bearingType` on every invocation of `updateLocation`. Instead `bearingType` should be  a stored property in `Puck2D` / `Puck3D` . The thing to take care of in that situation is that updating the `bearingType` results in an update of the store property in the `Puck`.

Reply:
I see your point, but that feels like it might add overhead as opposed to just passing it through. At the end the day, the switch statement won't change

Reply:
Hmm. It's not typical to pass this kind of configuration as part of doing some work.